### ISSUE TITLE
TCP_USER_TIMEOUT doesn't exist on solaris 11.4 

### ISSUE THREAD

**Issue by ciaranmcgirl** [celery/py-amqp#354]
in amqp/transport.py as part of the default TCP options we have 'TCP_USER_TIMEOUT': 1000 
this doesn't exist in solaris 11.4, which breaks celery connecting to rabbitmq

pip show celery
Name: celery
Version: 4.4.7

 pip show rabbitmq
Name: rabbitmq
Version: 0.2.0

```
celery amqp
-> connecting to amqp://guest:**@localhost:5672//.
Traceback (most recent call last):
<snip>
File "<snip>/celery/lib/python3.7/site-packages/amqp/transport.py", line 220, in _set_socket_options
    self.sock.setsockopt(SOL_TCP, opt, val)
OSError: [Errno 22] Invalid argument

```
adding the following to platform.py under the sunos elif resolves the issue
`KNOWN_TCP_OPTS.remove('TCP_USER_TIMEOUT')`
```

celery amqp
-> connecting to amqp://guest:**@localhost:5672//.
-> connected.
1>
```

**Comment by matusvalo** [comment]
Thank you @ciaranmcgirl for your report. Would you mind creating a PR fixing this issue?

**Comment by matusvalo** [comment]
Closing since Fixed by #355.
